packages/hurl/src/runner/result.rs

```diff
@@ -141,34 +143,45 @@ impl EntryResult {
         stdout: &mut Stdout,
         source_info: SourceInfo,
     ) -> Result<(), Error> {
-        let Some(call) = self.calls.last() else {
-            return Ok(());
-        };
-        // We check file access authorization for file output when a context dir has been given
-        if let Output::File(filename) = output {
-            if !context_dir.is_access_allowed(filename) {
-                let inner = RunnerError::UnauthorizedFileAccess {
-                    path: PathBuf::from(filename.clone()),
-                };
-                return Err(Error::new(source_info, inner, false));
+        if let Some(call) = self.calls.last() {
+            if let Output::File(filename) = output {
+                check_file_access(filename, context_dir, &source_info)?;
             }
-        }
-        let response = &call.response;
-        if self.compressed {
-            let bytes = match response.uncompress_body() {
-                Ok(bytes) => bytes,
-                Err(e) => {
-                    // TODO: pass a [`SourceInfo`] in case of error
-                    // We may pass a [`SourceInfo`] as a parameter of this method to make
-                    // a more accurate error (for instance a [`SourceInfo`] pointing at
-                    // `output: foo.bin`
-                    let source_info = SourceInfo::new(Pos::new(0, 0), Pos::new(0, 0));
-                    return Err(Error::new(source_info, e.into(), false));
+            let response = &call.response;
+            let body = if self.compressed {
+                match response.uncompress_body() {
+                    Ok(bytes) => bytes,
+                    Err(e) => return {
+                        // TODO: pass a [`SourceInfo`] in case of error
+                        // We may pass a [`SourceInfo`] as a parameter of this method to make
+                        // a more accurate error (for instance a [`SourceInfo`] pointing at
+                        // `output: foo.bin`
+                        let source_info = SourceInfo::new(Pos::new(0, 0), Pos::new(0, 0));
+                        return Err(Error::new(source_info, e.into(), false));
+                    },
                 }
+            } else {
+                response.body.clone()
             };
-            output.write(&bytes, stdout, Some(context_dir))
-        } else {
-            output.write(&response.body, stdout, Some(context_dir))
+
+            if body.len() > self.max_filesize {
+                output.write(&body, stdout, Some(context_dir))
+            }
         }
+        Ok(())
+    }
+}
+
+fn check_file_access(
+    filename: &str,
+    context_dir: &ContextDir,
+    source_info: &SourceInfo,
```

Comment:
`SourceInfo` implements `Copy` so this should bother clippy. You can run `cargo clippy`to check if everything is OK (indicated in the [CONTRIBUTING.md guide](https://github.com/Orange-OpenSource/hurl/blob/master/CONTRIBUTING.md)).